client/lib/discounts/index.js

```diff
@@ -5,12 +5,14 @@
  */
 import activeDiscounts from './active-discounts';
 
-function getDiscountByName( discountName ) {
+function getDiscountByName( discountName, endsAt ) {
 	const activeDiscount = activeDiscounts.find( function( discount ) {
 		if ( discountName !== discount.name ) {
 			return false;
 		}
 
+		if ( typeof endsAt !== 'undefined' ) discount.endsAt = endsAt;
```

Comment:
Should add curly braces and split it into multiple lines for better readability.
https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#blocks-and-curly-braces

Or, I think the following shorthand would work too:
`discount.endsAt = endsAt || discount.endsAt;`